read.go

```diff
@@ -269,10 +266,52 @@ func readNativeFrames(d dicomio.Reader, parsedData *Dataset, fc chan<- *frame.Fr
 
 	debug.Logf("readNativeFrames:\nRows: %d\nCols:%d\nFrames::%d\nBitsAlloc:%d\nSamplesPerPixel:%d", MustGetInts(rows.Value)[0], MustGetInts(cols.Value)[0], nFrames, bitsAllocated, samplesPerPixel)
 
+	bytesAllocated := bitsAllocated / 8
+	bytesRead = bytesAllocated * samplesPerPixel * pixelsPerFrame * nFrames
```

Comment:
Maybe we should call this variable bytesToRead in this context. By moving it earlier, I do think it changes what it was returning (previously presumably it was returning 0 for a lot of the return statements below), but I don't think this is used much anyway.

Reply:
When we parse the broken DICOM files, the library reads the data in the stream by `bytesToRead` , so it doesn't compare VL to `bytesToRead`  and returns an EOF error, which is not really useful.     

There is some case when the  length remaining bytes is smaller than  `bytesToRead`. So in general, this change improves better error message.